pkg/network/tracer/offsetguess/overlaps.go

```diff
@@ -0,0 +1,129 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package offsetguess
+
+import (
+	"unsafe"
+
+	"golang.org/x/exp/slices"
+)
+
+// offset, size tuple
+type offsetRange struct {
+	Offset uint64
```

Comment:
How do you distinguish between a genuine `0` here and the initial state? Maybe if this is a pointer you can?

Reply:
They are only checked in order, so it doesn't ever check a field that hasn't been guessed yet.